ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -445,14 +446,23 @@ public static boolean isCompactionTable(Properties tblProperties) {
   }
 
   /**
-   * Determine if a table is used during query based compaction.
+   * Determine if a table is used during query based compaction for CRUD tables.
    * @param parameters table properties map
    * @return true, if the parameters contains {@link AcidUtils#COMPACTOR_TABLE_PROPERTY}
    */
   public static boolean isCompactionTable(Map<String, String> parameters) {
     return Boolean.valueOf(parameters.getOrDefault(COMPACTOR_TABLE_PROPERTY, "false"));
   }
 
+  /**
+   * Determine if a table is used during query based compaction for MM insert-only tables.
+   * @param parameters table properties map
+   * @return true, if the parameters contains {@link AcidUtils#MM_COMPACTOR_TABLE_PROPERTY}
+   */
+  public static boolean isMmCompactionTable(Map<String, String> parameters) {
```

Comment:
Shouldn't isCompactionTable return true in both cases? Isn't it a problem the other places we use this util, that the mmCompactionTable are missed?

Reply:
isCompactionTable logically would be true for both full acid and mm tables, but until now we've only used it to mark tables used for compacting full acid tables. AFAIK we don't want to apply the operations we do on full acid compaction tables to mm compaction tables.
I could rename isCompactionTable() to isFullAcidCompactionTable() for easier reading, would that do?

Reply:
Done.